### PR TITLE
Debug invitation

### PR DESCRIPTION
**Pull request by qiyundai**
The invitations no longer kick the original owner of the trip out!

### CODE REVIEW

**Review by fredThem — CHANGES_REQUESTED** [review]
image
Screen Shot 2021-04-19 at 8 12 36 PM
Trip duplicated in views*

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -10,20 +10,20 @@ class ApplicationController < ActionController::Base
...
   def trips
-    @trips = policy_scope(Trip).order(created_at: :desc)
+    @trips = policy_scope(Trip).order(:start_date)
     if current_user
-      @my_trips = current_user.trips
+      @my_trips = []
       @trips.each do |trip|
         trip.invitations.each do |invitation|
-          @my_trips << invitation.trip if invitation.user == current_user
+          @my_trips << invitation.trip if trip.user == current_user ||invitation.user == current_user
```
Comment (fredThem) [inline app/controllers/application_controller.rb:18]: I think this here is the issue, maybe try find_or_create_by 
https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by
or something like that

**Review by fredThem — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (qiyundai) [reply app/controllers/application_controller.rb]: Sorry for the oversight. It's missing `unless @my_trips.include? trip`. Added in the new commit.

**Comment by qiyundai** [comment]
Also, this PR has a db:migrate to run.

**Pull request merged**